tests/admin_views/models.py

```diff
@@ -938,3 +938,14 @@ class ExplicitlyProvidedPK(models.Model):
 
 class ImplicitlyGeneratedPK(models.Model):
     name = models.IntegerField(unique=True)
+
+
+# Models for #25622
+class ReferencedByGenRel(models.Model):
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    object_id = models.PositiveIntegerField()
+    content_object = GenericForeignKey('content_type', 'object_id')
+
+
+class GenRelReference(models.Model):
+    references = GenericRelation(ReferencedByGenRel)
```

Comment:
Unfortunately there's no way to create an isolated test case without the creation of additional models.

I might have an idea to speed up the test suite on backend with slow table creation the other hand, it looks like many models defined around the test code base are never queried (that's the case here for both of them). I'll try to investigate if setting `managed = False` on them is worth the trouble. In the case here the tests pass with this optimisation.